python/hsfs/storage_connector.py

```diff
@@ -587,6 +588,51 @@ def prepare_spark(self, path: Optional[str] = None):
     def _get_path(self, sub_path: str):
         return os.path.join(self.path, sub_path)
 
+    def read(
+        self,
+        query: str = None,
+        data_format: str = None,
+        options: dict = {},
+        path: str = "",
+    ):
+
+        """Reads a path into a dataframe using the storage connector.
+        # Arguments
+            query: Not relevant for ADLS connectors.
+            data_format: The file format of the files to be read, e.g. `csv`, `parquet`.
+            options: Any additional key/value options to be passed to the ADLS connector.
+            path: Path within the bucket to be read. e.g. 'abfss://[container-name]@[account_name].dfs.core.windows.net/[path]'.
+            If no path is specified default path will be derived from connector.
+
+        # Returns
+            `DataFrame`.
+        """
+        return engine.get_instance().read(
+            self, data_format, options, self._validate_path(path)
+        )
+
+    def _validate_path(self, read_path: str) -> str:
+        ERROR_GEN2_PATH = "Not a valid ADLS path. For Gen2 connectors, path should follow format as 'abfss://[container-name]@[account_name].dfs.core.windows.net/[path]' "
+        ERROR_GEN1_PATH = "Not a valid ADLS path. For Gen1 connectors, path should follow format as 'adl://[account_name].azuredatalakestore.net' "
```

Comment:
these messages are quite similar, maybe we could save it as a template and just insert the parts that change (ex. `Gen2` and `'abfss://[container-name]@[account_name].dfs.core.windows.net/[path]'`)